### DISCUSSION TITLE
How can I add filename and line number to logging?

### DISCUSSION THREAD

**Discussion by parsley72** [gabime/spdlog#2389] [Q&A]
My previous logging code worked like this:
```
#define log(pri, fmt, ...) \
    syslog(pri, "[%s:%d] " fmt, __FILE__, __LINE__, ##__VA_ARGS__)

#define log_crit(fmt, ...) log(LOG_CRIT, fmt, ##__VA_ARGS__)
#define log_err(fmt, ...) log(LOG_ERR, fmt, ##__VA_ARGS__)
#define log_warn(fmt, ...) log(LOG_WARNING, fmt, ##__VA_ARGS__)
#define log_info(fmt, ...) log(LOG_INFO, fmt, ##__VA_ARGS__)
#define log_notice(fmt, ...) log(LOG_NOTICE, fmt, ##__VA_ARGS__)
#define log_debug(fmt, ...) log(LOG_DEBUG, fmt, ##__VA_ARGS__)
```
I've ported this to spdlog but I've lost the filename and line number. How can I add this back in? I see in the code there's a struct called `source_loc` but there aren't any examples.

**Comment by parsley72** [comment]
Just seen https://github.com/gabime/spdlog/wiki/0.-FAQ#source-information-do-not-appear-when-using-custom-format. But if I use `SPDLOG_INFO` then do I need to set `SPDLOG_ACTIVE_LEVEL` at compile time or can I still adjust the logging level dynamically?

**Reply by tt4g** [reply]
Since `SPDLOG_ACTIVE_LEVEL` is the macro, it can be set only at compile time.

**Reply by parsley72** [reply]
Obviously, but if the debug level is built in using `SPDLOG_ACTIVE_LEVEL` can I still limit it dynamically using `set_level()`?

**Reply by tt4g** [reply]
logging macros (such as `SPDLOG_ACTIVE_LEVEL`, `SPDLOG_INFO` and `SPDLOG_LOGGER_INFO`) and `set_level()` API  are evaluated at different times.

It would be more understandable to look at the definition of logging macros. These are macros that allow the preprocessor to remove unwanted logging statements.
See: https://github.com/gabime/spdlog/blob/v1.10.0/include/spdlog/spdlog.h#L278-L339

`set_level()` is the API for filtering log messages at runtime.

In other words, setting `SPDLOG_LEVEL_TRACE` to `SPDLOG_ACTIVE_LEVEL` will leave logging statements defined in all logging macros.
However, you can limit the log messages output by `set_level()`.
